tools/testing/e2e/test_utils.h

```diff
@@ -0,0 +1,154 @@
+#ifndef IREE_TESTING_COMMON_UTILS_H_
+#define IREE_TESTING_COMMON_UTILS_H_
+
+#include <stdio.h>
+
+#include "iree/base/api.h"
+#include "iree/base/internal/flags.h"
+#include "iree/hal/api.h"
+#include "iree/vm/api.h"
+
+IREE_FLAG(bool, require_exact_results, true,
+          "Requires floating point result elements to match exactly.");
+IREE_FLAG(
+    float, acceptable_fp_delta, 1e-5f,
+    "Maximum absolute difference allowed with inexact floating point results.");
+IREE_FLAG(
+    int32_t, max_elements_to_check, 10000,
+    "Maximum number of tensor elements to check for the given test. For larger "
+    "matrices, only every n-th element will be checked for some n chosed to "
+    "stay just under that threshold and to avoid being a divisor of the inner "
```

Comment:
Is this only running on matrices? Could update the comment to "buffers" (or "tensors", but the runtime just sees buffers)
```suggestion
    "Maximum number of tensor elements to check for the given test. For larger "
    "buffers, only every n-th element will be checked for some n chosen to "
    "stay just under that threshold and to avoid being a divisor of the inner "
```